ompi/mca/part/persist/part_persist.c

```diff
@@ -39,6 +47,492 @@ ompi_part_persist_t ompi_part_persist = {
     }
 };
 
+/**
+ * This is a helper function that frees a request. This requires ompi_part_persist.lock be held before calling.
+ */
+__opal_attribute_always_inline__ static inline int
```

Comment:
What's the point of having this function as `__always_inline__` when its single use is protected by a mutex and the function itself has many calls to release objects and memory ?